config/v1/0000_10_config-operator_01_infrastructure-CustomNoUpgrade.crd.yaml

```diff
@@ -72,7 +72,35 @@ spec:
                       type: object
                     baremetal:
                       description: BareMetal contains settings specific to the BareMetal platform.
+                      properties:
+                        apiServerInternalIPs:
+                          description: apiServerInternalIPs are the IP addresses to contact the Kubernetes API server that can be used by components inside the cluster, like kubelets using the infrastructure rather than Kubernetes networking. These are the IPs for a self-hosted load balancer in front of the API servers. In dual stack clusters this list contains two IP addresses. In single stack clusters a single IP address is expected. When omitted, values from the status.apiServerInternalIPs will be used. Once set, the list cannot be completely removed (but its second entry can).
+                          format: ip
+                          items:
+                            type: string
+                          maxItems: 2
+                          type: array
+                          x-kubernetes-list-type: set
+                        ingressIPs:
+                          description: ingressIPs are the external IPs which route to the default ingress controller. The IPs are suitable targets of a wildcard DNS record used to resolve default route host names. In dual stack clusters this list contains two IP addresses. In single stack clusters a single IP address is expected. When omitted, values from the status.ingressIPs will be used. Once set, the list cannot be completely removed (but its second entry can).
+                          format: ip
+                          items:
+                            type: string
+                          maxItems: 2
+                          type: array
+                          x-kubernetes-list-type: set
+                        machineNetworks:
```

Comment:
Is `machineNetworks` required once `apiServerInternalIPs` and `ingressIPs` are set? If yes, should that be documented/verified?

Reply:
No it is not because we need to support upgraded clusters. As the only source of this information is install-config.yaml to which Cluster Network Operator has no access, it can be only o/installer who sets this field during installation time.

So for newly installed 4.15 clusters we will populate this field and for clusters that originated in 4.14 or older this field will remain empty.